python_modules/libraries/dagster-sigma/dagster_sigma/resource.py

```diff
@@ -61,6 +65,9 @@ class SigmaOrganization(ConfigurableResource):
         default=False,
         description="Whether to warn rather than raise when lineage data cannot be fetched for an element.",
     )
+    translator: ResourceDependency[Type[DagsterSigmaTranslator]] = Field(
```

Comment:
hmmmm why are we making this a resource dependency? Seems like the translator should just be a vanilla object/property

Reply:
Right now all params on a resource must be either a config type or a nested resource, e.g.

```python
E                   dagster._core.errors.DagsterInvalidPythonicConfigDefinitionError:
E                   Error defining Dagster config class <class 'dagster_sigma.resource.SigmaOrganization'> on field 'translator'.
E                   Unable to resolve config type typing.Type[dagster_sigma.translator.DagsterSigmaTranslator] to a supported Dagster config type.
E
E
E                   This config type can be a:
E                       - Python primitive type
E                           - int, float, bool, str, list
E                       - A Python Dict or List type containing other valid types
E                       - Custom data classes extending dagster.Config
E                       - A Pydantic discriminated union type (https://docs.pydantic.dev/usage/types/#discriminated-unions-aka-tagged-unions)
E
E
E                   If this config type represents a resource dependency, its annotation must either:
E                       - Extend dagster.ConfigurableResource, dagster.ConfigurableIOManager, or
E                       - Be wrapped in a ResourceDependency annotation, e.g. ResourceDependency[Type]
```

The actual passed translator can be a plain-old-python-object, the `ResourceDependency` here just insists that it is not treated as config.